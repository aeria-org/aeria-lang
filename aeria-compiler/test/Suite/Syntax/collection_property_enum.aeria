collection CollectionPropertyEnumTest {
  properties {
    prop enum @values(["value1", "value2", "value3"])
  }
}
