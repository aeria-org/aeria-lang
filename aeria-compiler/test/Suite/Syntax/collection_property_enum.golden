{
  "kind": "Program",
  "collections": [
    {
      "properties": [
        {
          "type": {
            "kind": "PEnum"
          },
          "name": {
            "value": "prop",
            "kind": "PropertyName"
          },
          "kind": "Property",
          "attributes": [
            {
              "kind": "Attribute",
              "attributeValue": {
                "literal": {
                  "value": [
                    {
                      "value": "value1",
                      "kind": "LString"
                    },
                    {
                      "value": "value2",
                      "kind": "LString"
                    }
                    ,
                    {
                      "value": "value3",
                      "kind": "LString"
                    }
                  ],
                  "kind": "LArray"
                },
                "kind": "ALiteral"
              },
              "attributeName": {
                "value": "values",
                "kind": "AttributeName"
              }
            }
          ]
        }
      ],
      "name": {
        "value": "CollectionPropertyEnumTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
